### ISSUE TITLE
MacOS proxy and tuntap

### ISSUE THREAD

**Issue by resource-not-found-blank** [nicocha30/ligolo-ng#55]
Hello!
Thank you for this tool.
But I have a problem with creating a TUN/TAP interface on MacOS:
```
sudo ip tuntap add user username mode tun ligolo
 > Object "tuntap" is unknown, try "ip help"`
```
Is there a solution to this problem?

**Comment by drxadz** [comment]
I am facing the same issue. Is there any solution?

```
sudo ip tuntap add user mac mode tun ligolo                                   1 ↵
Object "tuntap" is unknown, try "ip help".
```

**Comment by Hexix23** [comment]
I could manage to use the utun interfaces although. Something is broken with these macos stuff

**Comment by nicocha30** [comment]
Ligolo-ng v0.6.1 adds support for MacOS

**Comment by njmulsqb** [comment]
@nicocha30 I am still unable to run the command `sudo ip tuntap add user username mode tun ligolo` first of all it returned ip command not found which I installed using `brew install iproute2mac` but the issue still exists.

<img width="462" alt="Image" src="https://github.com/user-attachments/assets/9510e2be-32a6-44ac-8fa1-023c8ccffe55" />

**Comment by njmulsqb** [comment]
Thank you @nicocha30 but it's still not working for me.

Here's what I have done:

1. I ran the server and agent using selfcert
2. I ran ` sudo ifconfig utun3 alias 3.3.3.3 255.255.255.0` and `sudo route add -net 172.16.1.0/24 -interface utun3` on macOS
3. In the ligolo session, I ran `tunnel_start --tun utun3` and it gave following error

<img width="875" alt="Image" src="https://github.com/user-attachments/assets/239a0d95-27e3-4d5f-b6cf-8feff869c007" />
